File: hello-world/php/stubs/Sample/Genio/Paypal/Com/geniosample.php
<?php
namespace Sample\Genio\Paypal\Com;

use PayPal\Core\PPBaseService;
use PayPal\Core\PPUtils;
use PayPal\Exception\PPTransformerException;
use Sample\Genio\Paypal\Com\SayHelloRequest;
use Sample\Genio\Paypal\Com\SayHelloResponse;
/**
* AUTO GENERATED service wrapper class
*/
class geniosample extends PPBaseService {

	private static $SERVICE_NAME = "geniosample";
	private static $SERVICE_VERSION = "";
	private static $SDK_NAME    = "";
	private static $SDK_VERSION = "";

	public function __construct() {
		parent::__construct('geniosample', 'SOAP');
	}

	/**
	* 
	* @param Sample\Genio\Paypal\Com\SayHelloRequest $sayHelloRequest
	* @param PayPal\Core\PPApiContext $apiContext
	* @return Sample\Genio\Paypal\Com\SayHelloResponse
	*/
	public function sayHello( $sayHelloRequest, $apiContext) {
		
		$apiContext->addHttpHeader("SOAPAction", '"uri:com.paypal.genio.sample/sayHello"');
	
		$handlers = array(
			new \PayPal\Handler\GenericSoapHandler($this->xmlNamespacePrefixProvider()),
		);
		$resp = $this->call('geniosample', 'sayHello',  $sayHelloRequest, $apiContext, $handlers);
		
		$ret = new SayHelloResponse();
		$ret->init(PPUtils::xmlToArray($resp));
		return $ret;
	}



	/**
	 * fucntion with namespaces and corresponding prefixes used in SOAP request serialization
	 */
	public function xmlNamespacePrefixProvider(){

			$namespace = "";
			$namespace .= 'xmlns:xml="http://www.w3.org/XML/1998/namespace" ';
			$namespace .= 'xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" ';
			$namespace .= 'xmlns:soap="http://schemas.xmlsoap.org/wsdl/soap/" ';
			$namespace .= 'xmlns:tns="http://com.paypal.genio.sample" ';
			$namespace .= 'xmlns:SOAP-ENC="http://schemas.xmlsoap.org/soap/encoding/" ';
			$namespace .= 'xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" ';
			$namespace .= 'xmlns:xsd="http://www.w3.org/2001/XMLSchema" ';
			$namespace .= 'xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" ';

			return $namespace;
	}
}

File: hello-world/java/src/main/java/com/paypal/genio/sample/Geniosample.java
package com.paypal.genio.sample;

import com.paypal.core.APICallPreHandler;
import com.paypal.core.BaseAPIContext;
import com.paypal.core.BaseService;
import com.paypal.core.DefaultSOAPAPICallHandler;
import com.paypal.core.DefaultSOAPAPICallHandler.XmlNamespaceProvider;
import com.paypal.exception.*;
import com.paypal.genio.sample.SayHelloRequest;
import com.paypal.genio.sample.SayHelloResponse;
import com.paypal.sdk.exceptions.*;
import java.io.*;
import java.util.HashMap;
import java.util.Map;
import java.util.Properties;
import javax.xml.parsers.DocumentBuilderFactory;
import javax.xml.parsers.ParserConfigurationException;
import javax.xml.xpath.XPathConstants;
import javax.xml.xpath.XPathExpressionException;
import javax.xml.xpath.XPathFactory;
import org.w3c.dom.Node;
import org.xml.sax.InputSource;
import org.xml.sax.SAXException;

public class Geniosample extends BaseService {

	static {
		DefaultSOAPAPICallHandler.setXmlNamespaceProvider(new XmlNamespacePrefixProvider());
	}

	// Service Version
	public static final String SERVICE_VERSION = "";

	// Service Name
	public static final String SERVICE_NAME = "geniosample";

	//SDK Name
	private static final String SDK_NAME = "";

	//SDK Version
	private static final String SDK_VERSION = "";

	/**
	 * Default <code>Geniosample</code> Constructor.
	 * Initializes the SDK system with the default configuration file named
	 * 'sdk_config.properties' found in the class-path
	 */
	public Geniosample() {
		super.initializeToDefault();
	}

	/**
	 * <code>Geniosample</code> that uses the supplied
	 * {@link Properties} to initialize the SDK system. For values that the
	 * properties should holdconsult the documentation.
	 * The initialization context is maintained only for
	 * this instance of the class. Service level configuration.
	 *
	 * @param properties
	 *            {@link Properties} object
	 */
	public Geniosample(Properties properties) {
		super(properties);
	}

	/**
	 * <code>Geniosample</code> that uses the supplied
	 * {@link Map} to initialize the SDK system. For values that the
	 * configurationMap should hold consult the documentation.
	 * The initialization context is maintained only for
	 * this instance of the class. Service level configuration.
	 *
	 * @param configurationMap
	 *            {@link Map} object
	 */
	public Geniosample(Map<String, String> configurationMap) {
		super(configurationMap);
	}

	public SayHelloResponse sayHello(SayHelloRequest sayHelloRequest, BaseAPIContext baseAPIContext) throws SAXException, IOException, ParserConfigurationException, InvalidResponseDataException, HttpErrorException, ClientActionRequiredException, InvalidCredentialException, MissingCredentialException, OAuthException, SSLConfigurationException, InterruptedException {
		if (baseAPIContext == null) {
			throw new IllegalArgumentException("BaseAPIContext is null");
		}
		if (baseAPIContext.getHTTPHeaders() == null) {
			baseAPIContext.setHTTPHeaders(new HashMap<String, String>());
		}
		baseAPIContext.getHTTPHeaders().put("SOAPAction", "uri:sample.genio.paypal.com/sayHello");
		APICallPreHandler apiCallPreHandler = new DefaultSOAPAPICallHandler(sayHelloRequest, baseAPIContext, this.configurationMap, "sayHello");
		String response = call(apiCallPreHandler);
		InputSource inStream = new InputSource();
		inStream.setCharacterStream(new StringReader((String) response));
		try {
			Node node = (Node) XPathFactory.newInstance().newXPath().evaluate("Envelope/Body/SayHelloResponse", DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(inStream), XPathConstants.NODE);
			return new SayHelloResponse(node);
		} catch (XPathExpressionException exe) {
			throw new RuntimeException("Unable to parse response", exe);
		}
	}


	/**
	 * Implementation of DefaultSOAPAPICallHandler.XmlNamespaceProvider
	 */
	private static class XmlNamespacePrefixProvider implements XmlNamespaceProvider {

		private Map<String, String> namespaceMap;

		public XmlNamespacePrefixProvider() {
			namespaceMap = new HashMap<String, String>();
			namespaceMap.put("xml", "http://www.w3.org/XML/1998/namespace");
			namespaceMap.put("wsdl", "http://schemas.xmlsoap.org/wsdl/");
			namespaceMap.put("soap", "http://schemas.xmlsoap.org/wsdl/soap/");
			namespaceMap.put("tns", "http://sample.genio.paypal.com");
			namespaceMap.put("SOAP-ENC", "http://schemas.xmlsoap.org/soap/encoding/");
			namespaceMap.put("xsi", "http://www.w3.org/2001/XMLSchema-instance");
			namespaceMap.put("xsd", "http://www.w3.org/2001/XMLSchema");
			namespaceMap.put("SOAP-ENV", "http://schemas.xmlsoap.org/soap/envelope/");
		}

		public Map<String, String> getNamespaceMap() {
			return namespaceMap;
		}
	}

}
